internal/database/prune.go

```diff
@@ -1,24 +1,20 @@
 package database
 
 import (
-	"log/slog"
 	"time"
 )
 
-func DownsampleOverTime(interval time.Duration, database Database) error {
+func DownsampleOverTime(interval time.Duration, downsample_thresh time.Duration, delete_thresh time.Duration, database Database) error {
 	downsampleTicker := time.NewTicker(time.Duration(interval))
 
 	for range downsampleTicker.C {
-		err := database.Downsample(time.Now())
-
-		if err != nil {
-			slog.Error("Got error whilst downsampling", "err", err)
-		}
+		database.Downsample(-1 * downsample_thresh)
```

Comment:
NIT: Why not just `-downsample_thresh` here. (And loads of other places in this PR)